sdk/keyvault/azure-security-keyvault-administration/src/main/java/com/azure/security/keyvault/administration/implementation/models/SelectiveKeyRestoreOperation.java

```diff
@@ -0,0 +1,168 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+// Code generated by Microsoft (R) AutoRest Code Generator.
+
+package com.azure.security.keyvault.administration.implementation.models;
+
+import com.azure.core.annotation.Fluent;
+import com.fasterxml.jackson.annotation.JsonProperty;
+
+/** The SelectiveKeyRestoreOperation model. */
+@Fluent
+public final class SelectiveKeyRestoreOperation {
```

Comment:
There should be only a generic `RestoreOperation` for full- and selective-restore (similar to `BackupOperation` I mentoined above).

Reply:
I agree there should only be one operation for restore, however, AutoRest generated an [operation for the full restore](https://github.com/Azure/azure-sdk-for-java/blob/5ee4f513c77efeba8ecd2dff967c2fe564d37dd7/sdk/keyvault/azure-security-keyvault-administration/src/main/java/com/azure/security/keyvault/administration/implementation/KeyVaultBackupClientImpl.java#L112) and [one for the selective restore](https://github.com/Azure/azure-sdk-for-java/blob/5ee4f513c77efeba8ecd2dff967c2fe564d37dd7/sdk/keyvault/azure-security-keyvault-administration/src/main/java/com/azure/security/keyvault/administration/implementation/KeyVaultBackupClientImpl.java#L130) that use different return types. This also made me [write code](https://github.com/Azure/azure-sdk-for-java/blob/5ee4f513c77efeba8ecd2dff967c2fe564d37dd7/sdk/keyvault/azure-security-keyvault-administration/src/main/java/com/azure/security/keyvault/administration/KeyVaultBackupAsyncClient.java#L367) to convert one to the other for getting an operation's status.

I think we should ask the service team why they declared two different types for the restore operation (including header, response and parameter types) that have the same properties in their swagger. [Here](https://github.com/Azure/azure-sdk-for-java/tree/5ee4f513c77efeba8ecd2dff967c2fe564d37dd7/sdk/keyvault/azure-security-keyvault-administration/src/main/java/com/azure/security/keyvault/administration/implementation/models) is how all these classes look like in Java.